Professions: Blacksmithing & Mining

helloruhe opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
dont look at this i will be right back

<!--
A changelog where you can describe more complicated things to other developers and testers.
-->
## Developer changelog:
- Added this thing.
- Changed that thing.

<!--
Before the merge, we recommend you to do these tests with your branch.
-->
## Tests:
- [ ] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [ ] The mod takes less than 5.5 GB in the Task Manager (Windows)

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:

helloruhe committed:
initial trait addition

helloruhe committed:
icons, starting gui

Grekopithikos commented on common/traits/wc_professions_traits.txt:
```diff
@@ -0,0 +1,464 @@
+}
+
+profession_herbalism = {
+	icon = profession_herbalism
+	desc = {
+		first_valid = {
+			triggered_desc = {
+				trigger = {
+					NOT = { exists = this }
+				}
+				desc = trait_profession_herbalism_desc
+			}
+			desc = trait_profession_herbalism_character_desc
+		}
+	}
+	shown_in_ruler_designer = no
+	minimum_age = 14
+
+	group = profession
+	level = 1
+	flag = profession_trait
+	flag = profession_skinning
+	category = lifestyle
+
+	same_opinion = 10
+
+
+
+	track = {
+		20 = { }
+		40 = { }
+		60 = { }
+		80 = { }
+		100 = { }
+	}
+
+	compatibility = {
+		profession_alchemy = @pos_compat_high
+	}
+}
```
Do you think the already existing herbalist lifestyle could be used for that instead? 
Like taking all the flags, tracks and anything else you want to use from here and putting them into `lifestyle_herbalist`.
Because having both traits could be confusing.

helloruhe replied:
i am consiente this yes, but I'm gonna work on other professions first and then probably redo the herbalist lifestyle

helloruhe committed:
localization + remove leatherworking

helloruhe committed:
more loc, almost done w gui

helloruhe committed:
adding reagents

helloruhe committed:
reagents

helloruhe committed:
reagent discovery mechanic

edmund-h commented on common/scripted_effects/wc_professions_effects.txt:
```diff
@@ -0,0 +1,158 @@
+		set_variable = {
+			name = profession_2_oncd
+			value = yes
+		}
```
can we change these flags from cd / oncd to `cooldown`/`on_cooldown` it's just better to fully write things out

helloruhe replied:
i am removing gthis

edmund-h reviewed (commented):
just a comment to expand abbreviations into full words for core mechanics variables. we want maximum clarity

helloruhe committed:
removing and combination of some professions, creating story cycle

helloruhe committed:
more changes

helloruhe committed:
starting to overhaul gui + icons

helloruhe committed:
cleanup reagent discovery mechanic

helloruhe committed:
overhaul gui pt3

helloruhe committed:
smelting/selling reagents mechanic

edmund-h commented on common/scripted_effects/wc_professions_reagent_effects.txt:
```diff
@@ -0,0 +1,559 @@
+				limit = { reagent_rarity_is_rare_trigger = { REAGENT = $REAGENT$ } }
+				value = 5
+			}
+			if = {
```
shouldn't these be `else_if`s? a reagent can't be rare and uncommon at the same time can it?

edmund-h commented on common/scripted_triggers/wc_professions_reagent_triggers.txt:
```diff
@@ -0,0 +1,219 @@
+}
+
+#PROVINCE SCOPE
+province_has_reagent_trigger = {
```
i believe that we will have to separate this out into a separate effect. think of how absolutely enormous this will get once we have cooking and herbalism as well.

helloruhe committed:
gui options, split triggers and files for reading ability, fix province trigger

helloruhe committed:
note

helloruhe committed:
k

robmart committed:
Better being assignment (#1389)

* Switch all being assignment to using culture gfx

* Kvaldir assignment

* Remove demonic flag, it's not needed

* Remove age boost to plants, being_life already gets a boost

* Remove unneeded check

* Remove unneeded check

* Remove unneeded check

ElMariuso committed:
Evil Religious Attitude (#1400)

* Initial commit

* some grammatical changes to old god possessives

* revise english localization for dark pluralism doctrine

---------

Co-authored-by: Edmund <447ght@gmail.com>

robmart committed:
Lordaeron and Quel'thalas Arthas Questline (#1404)

* wc_arthas_story_50 tooltip fix

* a

* a

* lich king notification someone yoinked frostmourne

* aa

* aa

* Update wc_scourge_quelthalas_effects.txt

* Fire the effect

* Change commence_scourge_quelthalas_effect region check

* a

* Fix

* removing lordaeron notification from arthas_story, adding to separate decision

* remove quelthalas event chain starter from arthas story

* require frostmourne to start decision

* Create wc_scourge_quelthalas_decisions.txt

* reduce requirements

* event update

* loc fix

* decision update

* test

* additional scope

* Update wc_scourge_quelthalas_events.txt

* event two

* offer to quelthalas vassals

* Update wc_scourge_quelthalas_events.txt

* Change Arthas inheritance and order of Quel events

* Revamp Lordaeron inheritance again

* Add pressed claim to e_lordaeron

* prince dark'khan actually betrays quelthalas most of the time

* localisation for fealty events

* Update wc_scourge_quelthalas_events.txt

* End war properly and fix some errors

* scourge invasion of quelthalas cb

* event war update

* fix

* war cb

* Lordaeron always gets broken

* Calia always gets e_lordaeron claim

* war tweaks

* End the correct war

* Turn Lordaeron undead

* conversion from sc_scourge_qualthalas_effect to event chain

* Turn conquered vassals undead

* Better handling of quel war victory

* Turn *everyone* undead

* Also applies to Lordaeron

* Modifier to block union

* Inheritable union block modifier

* Block Lor'themar from union on later start date

* San'layn High Elf variant

* Should be Darkfallen

* darkfallen loc

* big flavour text

* notification events

* Update wc_scourge_quelthalas_events.txt

* Make sure no foreign rulers are made undead

* Fix

* Prevent characters with plot armor from being raised

* Add missing equal signs

* sylvanas banshee event

* Fix Sylvanas event

* Frostmourne is not instantly maxed

* Union block is removed correctly

* Equip Frostmourne automatically

* oh no

* Convert recipient properly

* Union block modifier loc

* sylvanas and arthas perspective

* Basic corrupted sunwell and decision

* book of medivh

* Don't plague lore character

* Better non-plaguing

* add loc for invasion of QT

* Change title change to conquest

* localization for scourge army QT invasion

* add some goals for future work on invasion of QT

* add another requirement to QT subject chance

* ALWAYS end the correct war

* add resurrection (kinda broken) [SCQT]

* change SCQT KT resurrect to use set_house for dyn

* Fix the res

* Add parents back

* Sets dummy parents for purged characters

* Fix errors

* Transfer children of resurrected characters

* Fix Kel'Thuzad trigger and appearance

* Remove impotent trait from undead

* Tiny simplification

* Another small fix

* fix SCQT edge cases

* no more removing arthas as regent!

* fix scqt.1001

* book works for non dlc holders

* purple book is nicer (wish we had orange)

* sylvanas loc

* decision loc

* event chain loc

* Remove the second Kel'thuzad

* equip_artifact_to_owner_replace loc

* Generalize usurp_lordaeron_effect and give it loc

* ranger general now dies with correct grammar

* i made second dot

* Update wc_scourge_quelthalas_events_l_english.yml

* turn_realm_undead_effect effect loc

* More bullet points

* Fixed event option loc

* Banshee/Spectre custom loc

* Fix Quel'thalas borders after war

* Give converted elves event army and fix small errors

* Give Quel'thalas to an undead Elf

* missed some gendered localisation

* Add frostmourne to triggers

* Remove special building from visible triggers

* loc update

* weight modifiers

* Fix awkward sunwell building trigger loc

* Frostmourne trigger loc

* Darkfallen icon

* Fix remaining errors

---------

Co-authored-by: Raged50 <raged50@hotmail.com>
Co-authored-by: Edmund <447ght@gmail.com>

Chomski2 committed:
Orc armor (#1356)

* Orc armor

* Male blendshapes

Male blendshapes for orc armor

* Some fixes

* work work

Adjusted male helmet
Added female helmet
Adjusted male blendshapes

* Orc armor

* Orc armor

* Female Blendshapes

* Blendshape more like lameshape

* I hate blendshapes and titts

* First test for textures

* CoA on orc armor

* Final (?) male armor and textures

Rigging and texture of the male armor are finish for now

* Female model

* drakkari dont wear orc clothing

* Blendshape crash game

* No more crashes

No idea why

* Fixed ass

* Helmet coa and some fixes

* breast bandage

* bandage added to clothing

* No more orc tits

* Less helmets

Theres a 60% the character use a helmet

* Randomness working properly

* No major orc wears helmet

* Consistent code

---------

Co-authored-by: Raged50 <raged50@hotmail.com>

helloruhe committed:
Update todo

not workon on professions at the moment, but this commit will tell me what needs to be worked on when  i pick it up again

helloruhe committed:
remove scrapped reagent

helloruhe committed:
added missing triggers

helloruhe committed:
todo

helloruhe commented:
will work on this later
